src/skmultiflow/meta/learn_pp.py

```diff
@@ -177,8 +177,8 @@ def partial_fit(self, X, y, classes=None, sample_weight=None):
         return self
 
     def __fit_batch(self, X, y):
-        ensemble = [copy.deepcopy(self.base_estimator) for _ in range(self.n_estimators)]
-        normalized_errors = [1.0 for _ in range(self.n_estimators)]
+        ensemble = []
```

Comment:
What is the rationale behind this change?

Reply:
Originally, the ensemble algorithm is designed to train all of its members and combine their predictions considering the observed normalized errors. However, when training the classifiers, if the observed prediction error shrinks to zero before all learners are fit, the error normalization results in a `ValueError: probabilities contain NaN` message. This happens because the error normalization factor is zero, so the resulting division (for weight normalization) is `NaN`. I created a [new test](https://github.com/scikit-multiflow/scikit-multiflow/blob/18b8f518f067e4739d850ddbd91e30fe5e7c3adb/tests/meta/test_learn_pp.py#L58-L100) to encompass this situation.

In the modified version, ensemble members are incrementally added until the maximum number of learner is reached or the observed error is zero.

Reply:
This is a valid concern and a valid implementation-wise solution. However, my concern is that this is not part of the original algorithm. We could add a note in the code (basically what you wrote above). I would assume this only happens in rare occasions.

Reply:
Good idea @jacobmontiel and a valid observation. I added a note to `LearnPPClassifier` in commit [c460875](https://github.com/scikit-multiflow/scikit-multiflow/pull/223/commits/c460875b27a8aef9a70a33c14e94d82b54f8c045). In the note, I state the reason for the early stop mechanism and also mention that this situation is very specific. Thank you for pointing that out.